WordPress/Classes/ViewRelated/Blog/SharingViewController.m

```diff
@@ -95,7 +107,27 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
         case SharingPublicize: {
             Publicizer *publicizer = self.publicizeServices[indexPath.row];
             cell.textLabel.text = publicizer.label;
-            } break;
+
+            UIButton *button = nil;
+            if ([cell.accessoryView isKindOfClass:[UIButton class]]) {
+                button = (UIButton *)cell.accessoryView;
+            } else {
+                button = [UIButton buttonWithType:UIButtonTypeSystem];
+                [button addTarget:self action:@selector(publicizeButtonTapped:) forControlEvents:UIControlEventTouchUpInside];
+                cell.accessoryView = button;
+            }
+            button.tag = indexPath.row;
+            NSString *title = nil;
+            if ([self.connectingService.service isEqualToString:publicizer.service]) {
+                title = NSLocalizedString(@"Connecting…", @"Button title while a Publicize service is connecting");
+            } else if (publicizer.isConnected) {
+                title = NSLocalizedString(@"Disconnect", @"Button title to disconnect a Publicize service");
+            } else {
+                title = NSLocalizedString(@"Connect", @"Button title to connect a Publicize service");
+            }
+            [button setTitle:title forState:UIControlStateNormal];
+            [button sizeToFit];
```

Comment:
Sorry i didn't mention this before! came up while testing. (Suggestion!) maybe it'd be cleaner to use a simple `UITableViewCell` set with `UITableViewCellStyleValue1` (or a plain tableView with a right label as an accessory).

Reasons to do so are:
- No need, really, to use a button. We could handle taps over the entire row
- There's a weird animation being rendered on 'Title Switch', even though the cell is being refreshed with no animation.

(Just a comment / opinion). You've mentioned elsewhere something about a placeholder. If that's the case, and this view is up for refactor, please, feel free to disregard sir!